api/src/test/java/org/openmrs/api/impl/PatientServiceImplTest.java

```diff
@@ -271,28 +329,61 @@ public void getPatientIdentifierTypes_shouldReturnEmptyListGivenDaoReturnsNull()
 		assertEquals(0, actualIdentifierTypes.size());
 	}
 
+	@Test
+	public void exitFromCare_shouldFailIfGivenPatientIsNull() {
+		
+		when((messageSourceServiceMock).getMessage(eq("Patient.invalid.care"), any(), any()))
+			.thenReturn("localized error message");
+		
+		Exception thrown = assertThrows(APIException.class,
+			() -> patientService.exitFromCare(null, newDate, concept));
+		assertThat(thrown.getMessage(), is("localized error message"));
+	}
+
+	@Test
+	public void exitFromCare_shouldFailIfGivenDateExitedIsNull() {
+
+		when((messageSourceServiceMock).getMessage(eq("Patient.no.valid.dateExited"), any(), any()))
+			.thenReturn("localized error message");
+		
+		Exception thrown = assertThrows(APIException.class,
+			() -> patientService.exitFromCare(patient, null, concept));
+		assertThat(thrown.getMessage(), is("localized error message"));
+	}
+
+	@Test
+	public void exitFromCare_shouldFailIfGivenConceptIsNull() {
+
+		when((messageSourceServiceMock).getMessage(eq("Patient.no.valid.reasonForExit"), any(), any()))
+			.thenReturn("localized error message");
+		
+		Exception thrown = assertThrows(APIException.class,
+			() -> patientService.exitFromCare(patient, newDate, null));
```

Comment:
```suggestion
			() -> patientService.exitFromCare(new Patient(), new Date(), null));
```

Reply:
okay.... i'll do this... but my idea was to reuse the common code that i wrote (in before each test setup).....
By doing this we create two objects for patient, date and concept and use only one 😕

Reply:
Duplication in test code is not a bad thing. This improves readability. If you use a variable for patient, concept, date I might wonder is there something special about them? So I navigate to their instantiation just to see that no they are just `new Date()`, `new Patient()`, ... If I look at the suggested test I have everything I need to know in a line or two. I understand at one glance what case is tested.

This also reduces the amount of code that is needed in total.